test/integration/controller/admissionchecks/provisioning/provisioning_test.go

```diff
@@ -456,6 +438,127 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 			})
 		})
 
+		ginkgo.It("Should deactivate the workload when it's not Finished, and the ProvisioningRequest's condition is set to CapacityRevoked", func() {
+			// This happens only if the job allows retries and a user sets .spec.backOffLimit > 0
+			ginkgo.By("Setting the admission check to the workload", func() {
+				updatedWl := &kueue.Workload{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, updatedWl)).Should(gomega.Succeed())
+					util.SetWorkloadsAdmissionCheck(ctx, k8sClient, updatedWl, ac.Name, kueue.CheckStatePending, false)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Setting the quota reservation to the workload", func() {
+				updatedWl := &kueue.Workload{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, updatedWl)).Should(gomega.Succeed())
+					gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, updatedWl, admission)).To(gomega.Succeed())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Setting the ProvisioningRequeset as CapacityRevoked", func() {
+				createdRequest := &autoscaling.ProvisioningRequest{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, provReqKey, createdRequest)).Should(gomega.Succeed())
+					apimeta.SetStatusCondition(&createdRequest.Status.Conditions, metav1.Condition{
+						Type:   autoscaling.CapacityRevoked,
+						Status: metav1.ConditionTrue,
+						Reason: autoscaling.CapacityRevoked,
+					})
+					g.Expect(k8sClient.Status().Update(ctx, createdRequest)).Should(gomega.Succeed())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Checking if workload is deactivated and an event is emitted", func() {
```

Comment:
Check also the Evicted condition. And if the corresponding metric of the evicted workloads is incremented.

Reply:
Fixed, without checking the metric though as it would require a lot of changes in tests.

Reply:
It should be in a follow up at least.

However, I think if you set `active: false`, and let the `workload_controller` handle setting `WorkloadInactive` condition the metric will bump (see [here](https://github.com/kubernetes-sigs/kueue/blob/2b5184c3b1da933321b1c52795f4b2e04094c7ec/pkg/controller/core/workload_controller.go#L211))

Reply:
I've fixed this in this PR eventually